src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -486,6 +488,14 @@ public AgentInfo invoke(File nodeRoot, VirtualChannel virtualChannel) throws IOE
             } else {
                 binaryCompatible = false;
             }
+            String archType = "";
+            if (os == OsType.DARWIN) {
+                if (arch.contains("aarch") || arch.contains("arm")) {
```

Comment:
Without having any apple silicon machines to test on, I did my best to figure out what `os.arch` would actually contain

Reply:
You can look it up online.

Reply:
Yeah, i checked online and was pretty sure its `aarch64`, just wanted it to be known that I never actually tested it 😛